iis-wf/iis-wf-referenceextraction/src/main/java/eu/dnetlib/iis/wf/referenceextraction/softwareurl/CachedWebCrawlerJob.java

```diff
@@ -153,12 +153,21 @@ public static void main(String[] args) throws Exception {
             // store final results
             JavaRDD<Fault> faultsToBeStored = produceFaultToBeStored(toBeProcessed, returnedFromWebcrawlTuple._2);
             faultsToBeStored.cache();
-            
+
+            long entitiesReturnedFromCacheCount = entitiesReturnedFromCache.count();
+            long faultsReturnedFromCacheCount = inputJoinedWithCache
+                    .filter(x -> x._2._2.isPresent() && !x._2._2.get().isPresent())
+                    .count();
+            long processedEntitiesCount = webcrawledEntities.count();
+            long processedFaultsCount = faults.mapToPair(e -> new Tuple2<>(e.getInputObjectId(), 1))
+                    .join(inputByUrl)
+                    .count();
```

Comment:
I guess the count related comment could be applicable here as well. We should check whether it does induce the caching requirement.